### PR TITLE
fix(api): analytics module

### PR DESCRIPTION
**Pull request by scopsy**
### What changed? Why was the change needed?

Mis import of the analytics module, caused FE events to be dismissed.

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/analytics/analytics.module.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import { AnalyticsController } from './analytics.controller';
 import { HubspotIdentifyFormUsecase } from './usecases/hubspot-identify-form/hubspot-identify-form.usecase';
+import { SharedModule } from '../shared/shared.module';
 
 @Module({
-  imports: [HttpModule],
+  imports: [SharedModule, HttpModule],
   controllers: [AnalyticsController],
-  providers: [AnalyticsService, HubspotIdentifyFormUsecase],
+  providers: [HubspotIdentifyFormUsecase],
```
Comment (djabarovgeorge) [inline apps/api/src/app/analytics/analytics.module.ts:10]:
```suggestion
  providers: [analyticsService, HubspotIdentifyFormUsecase],
```

File: apps/api/src/app/analytics/analytics.module.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import { Module } from '@nestjs/common';
-import { AnalyticsService } from '@novu/application-generic';
 import { HttpModule } from '@nestjs/axios';
 import { AnalyticsController } from './analytics.controller';
 import { HubspotIdentifyFormUsecase } from './usecases/hubspot-identify-form/hubspot-identify-form.usecase';
+import { SharedModule } from '../shared/shared.module';
 
 @Module({
-  imports: [HttpModule],
+  imports: [SharedModule, HttpModule],
```
Comment (djabarovgeorge) [inline apps/api/src/app/analytics/analytics.module.ts:8]:
```suggestion
  imports: [HttpModule],
```

**Review by djabarovgeorge — COMMENTED** [review]
importing a custom `analyticsService` provider won't be enough?

File: apps/api/src/app/analytics/analytics.module.ts (hunk 1 above)
Comment (djabarovgeorge) [reply apps/api/src/app/analytics/analytics.module.ts:10]: feels like you were missing the process.env.SEGMENT_TOKEN injection? was there something else?

Comment (scopsy) [reply apps/api/src/app/analytics/analytics.module.ts:10]: It's already imported under the SharedModule, so I think I would love to have a single instance of it

Comment (scopsy) [reply apps/api/src/app/analytics/analytics.module.ts:10]: It's confusing, but that's how all of our service are exported from app generic

CleanShot 2024-10-13 at 17 41 30@2x

**Review by djabarovgeorge — APPROVED** [review]

File: apps/api/src/app/analytics/analytics.module.ts (hunk 1 above)
Comment (djabarovgeorge) [reply apps/api/src/app/analytics/analytics.module.ts:10]: got it, i was not sure if it would create 2 instances.

**Pull request merged**